tests/expected/global2.rs

```diff
@@ -0,0 +1,23 @@
+// cargo-deps: lazy_static
+extern crate lazy_static;
+use lazy_static::lazy_static;
+use std::collections::HashMap;
+use std::collections::HashSet;
+const code_0: i32 = 0;
+const code_1: i32 = 1;
+const code_a: &str = "a";
+const code_b: &str = "b";
+lazy_static! {
+    static ref l_b: HashSet<&'static str> = [code_a].iter().cloned().collect::<HashSet<_>>();
```

Comment:
`iter().cloned()` seems to work ok here, but is memory inefficient.

`into_iter()` seems to be more idiomatic code in my searches:

```
// cargo-deps: lazy_static
extern crate lazy_static;
use lazy_static::lazy_static;
use std::collections::HashMap;

lazy_static!{
    static ref MAP: HashMap<&'static str, i32> = vec![
        ("apple", 1),
        ("bear", 2),
        ("cat", 3),
    ].into_iter().collect();
}

fn main() {
    for (k, v) in MAP.iter() {
        println!("{} {}", k, v);
    }
}
```

Reply:
I'm ok with taking the code as-is and figure out how to make it more memory efficient later.

Also, the `iter()` in for loop needs to be brought back if we go this route (conditionally, only if the container is a static ref).

Reply:
`iter().cloned()` is an unrelated piece of the transpiler this PR is working on.  I suggest creating an issue about your suggestion here, as it is a good one, but should be looked at separately.

Re-adding `iter()` for specific container types will eventually be need.  The `strum` enums need `::iter`, although there is a issue in that project for adding `.iter()` iirc.  I'll look into this as one option to get this PR ready again.